### ISSUE TITLE
View Notebook Ouputs in Data Explorer

### ISSUE THREAD

**Issue by saulshanabrook** [jupyterlab/jupyterlab-data-explorer#16]
In order to support the previous integration we had with nteract's data explorer, we should be able to output a table mimetype from a notebook and view it with their data explorer by first registering it in ours. So we should try out enabling notebooks outputs as nested datasets.

I think they will be of the URL `file://filename.ipynb#cells/123`.

**Comment by saulshanabrook** [comment]
Talking with @tonyfast yesterday, he agreed that having notebook outputs as datasets should be a priority for the first release.

**Comment by tonyfast** [comment]
Yea I think there may be a best practice in treating notebooks as organizational data.  A `notebook` can have export commands based on `nbconvert`.

**Comment by saulshanabrook** [comment]
The high level idea is that when we resolve a notebook output like`file:///filename.ipynb#/cells/0/outputs/0` we first resolve the notebook file `file:///filename.ipynb` and get the `application/x.jupyterlab.notebook-context` ( of type `Context<INotebookModel>`) mimetype. From that, we can end up with the `application/x.jupyterlab.output-model` (`IOutputModel`) which has the changed signal and the data. And from that we can fill in the mimeTypes for the cell.

The goal is here to make sure that if an output is updated in a notebook, then the corresponding representation in the data registry is also updated. The easiest way to do that is to reuse the existing docregistry logic for grabbing the currently open context for a notebook or creating a new one.

**Comment by saulshanabrook** [comment]
We need to replicate this logic from `DocumentManager._createOrOpenDocument` to find a context pased on a path and factory name:

```typescript
    // Handle the load-from-disk case
    if (which === 'open') {
      // Use an existing context if available.
      context = this._findContext(path, factory.name) || null;
      if (!context) {
        context = this._createContext(path, factory, preference);
        // Populate the model, either from disk or a
        // model backend.
        ready = this._when.then(() => context.initialize(false));
      }
    } else if (which === 'create') {
      context = this._createContext(path, factory, preference);
      // Immediately save the contents to disk.
      ready = this._when.then(() => context.initialize(true));
    }
```

**Comment by saulshanabrook** [comment]
This was fixed by #32
